lib/tasks/publishing_api.rake

```diff
@@ -128,6 +128,45 @@ namespace :publishing_api do
     PublishingApiDocumentRepublishingWorker.new.perform(document.id)
   end
 
+  desc "Manually unpublish content with a redirect"
+  # This task is for unpublishing Whitehall managed content where
+  # Whitehall has forgotten it is managing the content (as it often
+  # does). Do not use this Rake task for content which Whitehall still
+  # has a record of.
+  namespace :unpublish_with_redirect do
+    task :dry_run, %i[content_id alternative_path locale] => :environment do |_, args|
+      args.with_defaults(locale: 'en')
+
+      document = Document.find_by(content_id: args[:content_id])
+      if document
+        puts "Document with this content ID exists: #{document}"
+        abort 'This Rake task is only for unknown content'
+      end
+
+      puts "Would send an unpublish request to the Publishing API for #{args[:content_id]} with:"
+      puts "  type 'redirect', locale: #{args[:locale]} and alternative_path #{args[:alternative_path].strip}"
+    end
+
+    task :real, %i[content_id alternative_path locale] => :environment do |_, args|
+      args.with_defaults(locale: 'en')
+
+      document = Document.find_by(content_id: args[:content_id])
+      if document
+        puts "Document with this content ID exists: #{document}"
+        abort 'This Rake task is only for unknown content'
+      end
```

Comment:
Could you reduce the duplication between the `:real` and `:dry_run` versions by extracting this out?

```ruby
def ensure_not_exists(content_id)
      document = Document.find_by(content_id: args[:content_id])
      if document
        puts "Document with this content ID exists: #{document}"
        abort 'This Rake task is only for unknown content'
      end
end

task :real, %i[content_id alternative_path locale] => :environment do |_, args|
    args.with_defaults(locale: 'en')
    ensure_not_exists(args[:content_id])
    # etc
end
```

Reply:
Yeah, I'll put that in.